Initial scripts for building Parcel and CSD

wangmiao1002 opened the pull request:
Kyuubi parcel and csd script

Commit:
Kyuubi integrated management service dependency project

pan3793 commented:
Thanks, I will take a look today.

pan3793 commented on bin/kyuubibulid.sh:
```diff
@@ -0,0 +1,80 @@
+CM_EXT_BRANCH=cm5-5.15.0
+kyuubi_version=1.6.1
+parcel_version=${kyuubi_version}-1
+kyuubi_package_url=https://archive.apache.org/dist/incubator/kyuubi/kyuubi-${kyuubi_version}-incubating/apache-kyuubi-${kyuubi_version}-incubating-bin.tgz
```
`-incubating` should be part of version, and since Kyuubi 1.7.0 has been released, we can move to 1.7.0 instead.

wangmiao1002 replied:
Ok, thank you for your reply. Please check again

Commit:
Change release version

pan3793 commented:
cc @bowenliang123 as well

pan3793 commented on parcel/kyuubi-parcel-src/meta/alternatives.json:
```diff
@@ -0,0 +1,8 @@
+{
+    "kbeeline": {
+      "destination": "/usr/bin/kbeeline",
```
To distinguish w/ hive beeline, maybe we should rename our scripts to `kyuubi-beeline`?
I saw that aliyun did such thing https://help.aliyun.com/document_detail/440233.html

And a background, CDH does not ship STS, so there is no conflict between hive beeline and spark beeline

cc @turboFei @yaooqinn

pan3793 commented on csd/kyuubi-csd-src/scripts/control.sh:
```diff
@@ -0,0 +1,134 @@
+    if [[ -z ${YARN_CONF_DIR} ]]; then
+  	KYUUBI_CLASSPATH="${KYUUBI_JAR_DIR}/*:${KYUUBI_CONF_DIR}:${HADOOP_CONF_DIR}"
+    else
+  	KYUUBI_CLASSPATH="${KYUUBI_JAR_DIR}/*:${KYUUBI_CONF_DIR}:${HADOOP_CONF_DIR}:${YARN_CONF_DIR}"
```
the whole scripts is kind of hacky, `bin/kyuubi run` can run KyuubiServer in foreground, have you tried this way?

wangmiao1002 replied:
No, at that time, the test used bin/kyuubi run to exit the process directly, and other processes were executing, for fear that the supervisor would not work, so this method was not used; Let me modify the test and use bin/kyuubi run

wangmiao1002 replied:
There are still some problems with using bin/kyuubi script, kyuubi_ HOME, kyuubi-env.sh and other parameters in the script are set. cm is automatically generated according to the page options, especially the configuration file under conf. If the conf generated by cm is placed under kyuubi/conf in parcels, there are still some permission problems to debug. Please see if you need to do this

pan3793 commented on csd/kyuubi-csd-src/descriptor/service.sdl:
```diff
@@ -0,0 +1,157 @@
+    "configWriter" : {
+       "peerConfigGenerators" : [
+          {
+            "filename": "conf/server.properties",
```
what's this?

wangmiao1002 replied:
It is used to generate client configuration, such as hive's hive-site.xml, which is saved on the gateway machine. The main reason for adding here is that there will be corresponding prompts to modify the configuration on the cm page. If you delete it, you will not be prompted. If you have any good suggestions, save some information in it

pan3793 commented on csd/kyuubi-csd-src/scripts/control.sh:
```diff
@@ -0,0 +1,134 @@
+function log {
+  timestamp=$(date)
+  echo "$timestamp: $1"       #stdout
+  echo "$timestamp: $1" 1>&2; #stderr
```
why log to both stdout and stderr?

wangmiao1002 replied:
Cm will generate stdout and stderr under each process. After the script is adjusted, the output of kyuubi will be directly used for printing. Can this output be removed

Commit:
1.Modify script startup mode
2.Modify the beeline name
3.Delete the added content example of kyuubi-env.sh and restore the original format of kyuubi

pan3793 commented on csd/kyuubi-csd-src/scripts/control.sh:
```diff
@@ -0,0 +1,72 @@
+    echo appender.console.target=SYSTEM_ERR >> conf/log4j2.properties
+    echo appender.console.layout.type=PatternLayout >> conf/log4j2.properties
+    echo appender.console.layout.pattern=%d{yy/MM/dd HH:mm:ss} %-5p %c{2}: [%t]: %m%n >> conf/log4j2.properties
+    exec ${KYUUBI_HOME}/bin/kyuubi run
```
https://github.com/apache/kyuubi/pull/4494 should make it work

pan3793 commented on csd/kyuubi-csd-src/descriptor/service.sdl:
```diff
@@ -0,0 +1,157 @@
+          "start"
+        ],
+      "environmentVariables":{
+                    "KYUUBI_WORK_DIR":"${kyuubi.session.work.path}",
```
```suggestion
                    "KYUUBI_WORK_DIR_ROOT":"${kyuubi_work_dir_root}",
```

pan3793 committed:
Update csd/kyuubi-csd-src/descriptor/service.sdl

pan3793 commented on csd/kyuubi-csd-src/descriptor/service.sdl:
```diff
@@ -0,0 +1,157 @@
+          "name": "kyuubi.session.work.path",
+          "label": "kyuubi.session.work.path",
+          "configName": "kyuubi.session.work.path",
+          "description": "Engines will be shared in different levels",
```
```suggestion
          "name": "kyuubi_work_dir_root",
          "label": "kyuubi_work_dir_root",
          "configName": "kyuubi_work_dir_root",
          "description": "Root directory for launching sql engine applications.",
```

pan3793 committed:
Update csd/kyuubi-csd-src/descriptor/service.sdl

pan3793 commented on csd/kyuubi-csd-src/descriptor/service.sdl:
```diff
@@ -0,0 +1,157 @@
+        },
+		{
+          "name": "kyuubi.session.engine.idle.timeout",
+          "label": "Doris Frontend query Port",
```
```suggestion
          "label": "kyuubi.session.engine.idle.timeout",
```

pan3793 committed:
Update csd/kyuubi-csd-src/descriptor/service.sdl

pan3793 commented on parcel/kyuubi-parcel-src/meta/alternatives.json:
```diff
@@ -0,0 +1,8 @@
+{
+    "kyuubi-beeline": {
+      "destination": "/usr/bin/kyuubi-beeline",
+      "source": "bin/beeline",
```
this soft link does not work, will fix later

pan3793 commented:
@wangmiao1002 thanks for making this great job! I think it's fine to merge first and do other improvements in follow ups.
